core/data.go

```diff
@@ -218,10 +226,10 @@ func (b Bundle) Deserialize(data []byte) (Bundle, error) {
 	}
 	// Parse metadata
 	meta := binary.LittleEndian.Uint64(data)
-	if (meta & 0xFF) != GnarkBundleEncodingFormat {
+	if (meta >> (NumBundleHeaderBits - NumBundleEncodingFormatBits)) != GnarkBundleEncodingFormat {
 		return nil, errors.New("invalid bundle data encoding format")
 	}
-	chunkLen := meta >> 8
+	chunkLen := meta << NumBundleEncodingFormatBits >> NumBundleEncodingFormatBits
```

Comment:
would be good to add (), or comment on the precedence on the binary operations are from left to right